# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/ia2/ia2AccessibleTableCell.cpp
# Commit: 6d0f4821ddcf
# Full Hash: 6d0f4821ddcfca1b59e4db3befc3ec9c12335891
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-03-23 09:49:32
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r d82660fb5408 -r 6d0f4821ddcf accessible/windows/ia2/ia2AccessibleTableCell.cpp
--- a/accessible/windows/ia2/ia2AccessibleTableCell.cpp	Wed Mar 23 04:00:25 2022 +0000
+++ b/accessible/windows/ia2/ia2AccessibleTableCell.cpp	Wed Mar 23 04:00:26 2022 +0000
@@ -74,7 +74,7 @@
   TableCellAccessible* tableCell = CellAcc();
   if (!tableCell) return CO_E_OBJNOTCONNECTED;
 
-  AutoTArray<LocalAccessible*, 10> cells;
+  AutoTArray<Accessible*, 10> cells;
   tableCell->ColHeaderCells(&cells);
 
   *aNColumnHeaderCells = cells.Length();
@@ -84,9 +84,7 @@
   if (!*aCellAccessibles) return E_OUTOFMEMORY;
 
   for (uint32_t i = 0; i < cells.Length(); i++) {
-    AccessibleWrap* cell = static_cast<AccessibleWrap*>(cells[i]);
-    RefPtr<IAccessible> iaCell;
-    cell->GetNativeInterface(getter_AddRefs(iaCell));
+    RefPtr<IAccessible> iaCell = MsaaAccessible::GetFrom(cells[i]);
     iaCell.forget(&(*aCellAccessibles)[i]);
   }
 
@@ -127,7 +125,7 @@
   TableCellAccessible* tableCell = CellAcc();
   if (!tableCell) return CO_E_OBJNOTCONNECTED;
 
-  AutoTArray<LocalAccessible*, 10> cells;
+  AutoTArray<Accessible*, 10> cells;
   tableCell->RowHeaderCells(&cells);
 
   *aNRowHeaderCells = cells.Length();
@@ -136,9 +134,7 @@
   if (!*aCellAccessibles) return E_OUTOFMEMORY;
 
   for (uint32_t i = 0; i < cells.Length(); i++) {
-    AccessibleWrap* cell = static_cast<AccessibleWrap*>(cells[i]);
-    RefPtr<IAccessible> iaCell;
-    cell->GetNativeInterface(getter_AddRefs(iaCell));
+    RefPtr<IAccessible> iaCell = MsaaAccessible::GetFrom(cells[i]);
     iaCell.forget(&(*aCellAccessibles)[i]);
   }
 